realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -282,6 +284,46 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread until new changes to the Realm are available or {@link #stopWaitForChange}
+     * is called from another thread. Once stopped waiting by calling {@link #stopWaitForChange},
+     * the Realm will <b>never</b> wait again for changes.
+     *
+     * @return {@code true} If the Realm was updated to the latest version, {@code false} if it was
+     * cancelled by calling {@link #stopWaitForChange} once.
+     * @throws IllegalStateException if attempting to wait within a transaction or a Looper thread.
+     */
+    public boolean waitForChange() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot wait for changes inside of a transaction.");
+        }
+        if (Looper.myLooper() != null) {
+            throw new IllegalStateException("Cannot wait for changes inside a Looper thread. Use RealmChangeListeners instead.");
+        }
+        boolean hasChanged = sharedGroupManager.getSharedGroup().waitForChange();
+        if (hasChanged) {
+            // Since this Realm instance has been waiting for change, advance realm & refresh realm.
+            sharedGroupManager.advanceRead();
+            handlerController.refreshSynchronousTableViews();
+        }
+        return hasChanged;
+    }
+
+    /**
+     * Makes {@link #waitForChange()} return {@code false} immediately. This method is threadsafe
```

Comment:
It would rephrase this to:

```
Makes any current {@link #waitForChange()} return {@code false} immediately. If a Realm is not waiting. all future calls to waitForChange will immediately return false as well.
<p>
This method is threadsafe and should _only_ be called from another thread than the one that called waitForChange.
```